core/lib/include/xfa.h

```diff
@@ -138,6 +141,13 @@ _Pragma("GCC diagnostic ignored \"-Warray-bounds\"")
  *
  *     XFA(driver_params, 0) driver_params_t _onboard = { .pin=42 };
  *
+ * @warning `driver_params_t` struct must be aligned at multiples of the word
```

Comment:
@maribu I've reworded this a bit to be clearer w.r.t. larger-than-wordsize alignments.